tests/functional/test_bad_url.py

```diff
@@ -0,0 +1,15 @@
+# test the error message returned by pip when
+# a bad "file:" URL is passed to it.
+
+from typing import Any
+
+
+def test_filenotfound_error_message(script: Any) -> None:
+    # Test the error message returned when using a bad 'file:' URL.
+    # make pip to fail and get an error message
+    # by running "pip install -r file:nonexistent_file"
+    proc = script.pip("install", "-r", "file:unexistent_file", expect_error=True)
+    assert proc.returncode == 1
+    file = "file:///unexistent_file"
+    expect = f"ERROR: 404 Client Error: FileNotFoundError for url: {file}"
```

Comment:
Any reason `file` needs to be a separate variable?

```suggestion
    expect = "ERROR: 404 Client Error: FileNotFoundError for url: file:///unexistent_file"
```

Reply:
Because `flake8` will fail, since that line would have 90 characters:

```python
>>>len('    expect = "ERROR: 404 Client Error: FileNotFoundError for url: file:///unexistent_file"')
90
```

I originally tried to follow your suggestion, but then `flake8` crashed. Can we do something else to avoid that long line, but removing the `file` variable?

Reply:
Does this work?

```python
expect = (
    "ERROR: 404 Client Error: FileNotFoundError for url: "
    "file:///unexistent_file"
)
```

Reply:
As I understand, `flake8` only supports less than 80 lines on our code, right?

Reply:
> Does this work?
> 
> ```python
> expect = (
>     "ERROR: 404 Client Error: FileNotFoundError for url: "
>     "file:///unexistent_file"
> )
> ```

Oh, great, thank you!

Unfortunately, I will have liimited time today (it's early Wednesday here), but maybe tomorrow I can commit your suggestion :smile:.

Reply:
@uranusjr I made your suggestion. Review it whenever you want.